Don't fail publishing if npm log output is too long...?

FUI repo recently had a failed publish which kind of looks like it was because the npm publish log output was too long...?

In the partial log output below, note how it cuts off in the middle of this last filename line:
`2020-03-26T12:28:05.0490036Z npm notice 1.2kB   dist/charting/dist-storybook/react-syntax-highlighter_lnpm ERR! code ELIFECYCLE`

```
2020-03-26T12:24:56.2663729Z Publishing - @fluentui/react-focus@7.1.11
2020-03-26T12:24:56.2665063Z publish command: publish --registry https://registry.npmjs.org/ --tag latest --//registry.npmjs.org/:_authToken=*** --access public
2020-03-26T12:25:07.8710529Z Published!
2020-03-26T12:25:07.8712204Z Publishing - @fluentui/examples@0.1.22
2020-03-26T12:25:07.8719336Z publish command: publish --registry https://registry.npmjs.org/ --tag latest --//registry.npmjs.org/:_authToken=*** --access public
2020-03-26T12:25:13.2018152Z Published!
2020-03-26T12:25:13.2020090Z Publishing - office-ui-fabric-react@7.105.4
2020-03-26T12:25:13.2021184Z publish command: publish --registry https://registry.npmjs.org/ --tag latest --//registry.npmjs.org/:_authToken=***
2020-03-26T12:25:42.6925191Z Published!
2020-03-26T12:25:42.6929325Z Publishing - @fluentui/react@7.105.5
2020-03-26T12:25:42.6930701Z publish command: publish --registry https://registry.npmjs.org/ --tag latest --//registry.npmjs.org/:_authToken=*** --access public
2020-03-26T12:25:51.4075014Z Published!
2020-03-26T12:25:51.4077026Z Publishing - @uifabric/fabric-website-resources@7.6.26
2020-03-26T12:25:51.4085556Z publish command: publish --registry https://registry.npmjs.org/ --tag latest --//registry.npmjs.org/:_authToken=*** --access public
2020-03-26T12:26:00.2080418Z Published!
2020-03-26T12:26:00.2082353Z Publishing - @uifabric/fabric-website@7.10.13
2020-03-26T12:26:00.2084095Z publish command: publish --registry https://registry.npmjs.org/ --tag latest --//registry.npmjs.org/:_authToken=*** --access public
2020-03-26T12:26:19.6458975Z Published!
2020-03-26T12:26:19.6460648Z Publishing - @uifabric/pr-deploy-site@7.0.45
2020-03-26T12:26:19.6462066Z publish command: publish --registry https://registry.npmjs.org/ --tag latest --//registry.npmjs.org/:_authToken=*** --access public
2020-03-26T12:28:04.8720379Z Something went wrong with the publish! Manually update these package and versions:
2020-03-26T12:28:04.8723443Z - @fluentui/react-focus@7.1.11
2020-03-26T12:28:04.8724293Z - @fluentui/examples@0.1.22
2020-03-26T12:28:04.8725107Z - office-ui-fabric-react@7.105.4
2020-03-26T12:28:04.8725729Z - @fluentui/react@7.105.5
2020-03-26T12:28:04.8726356Z - @uifabric/fabric-website-resources@7.6.26
2020-03-26T12:28:04.8736267Z - @uifabric/fabric-website@7.10.13
2020-03-26T12:28:04.8737520Z - @uifabric/pr-deploy-site@7.0.45
2020-03-26T12:28:04.8738373Z - @uifabric/api-docs@7.2.41
......
2020-03-26T12:28:04.8754293Z npm notice 
2020-03-26T12:28:04.8755005Z npm notice ðŸ“¦  @uifabric/pr-deploy-site@7.0.45
2020-03-26T12:28:04.8755811Z npm notice === Tarball Contents === 
2020-03-26T12:28:04.8756841Z npm notice 463.4kB dist/fabric-website-resources/dist/css/fabric-11.0.0.scoped.css                                                                               
2020-03-26T12:28:04.8758453Z npm notice 463.4kB dist/office-ui-fabric-react/css/fabric-11.0.0.scoped.css                                                                                      
...... many files later .......                       
2020-03-26T12:28:05.0489196Z npm notice 2.2kB   dist/todo-app/dist-storybook/react-syntax-highlighter_languages_highlight_makefile.a2aa69fba30a18ba7bae.bundle.js                             
2020-03-26T12:28:05.0490036Z npm notice 1.2kB   dist/charting/dist-storybook/react-syntax-highlighter_lnpm ERR! code ELIFECYCLE
2020-03-26T12:28:05.0490447Z npm ERR! errno 1
2020-03-26T12:28:05.0491397Z npm ERR! office-ui-fabric-react-repo@7.0.0 publish:beachball: `beachball publish --scope "!packages/fluentui/*" "-b" "origin/master" "-n" "***" "--access" "public" "-y"`
2020-03-26T12:28:05.0491999Z npm ERR! Exit status 1
2020-03-26T12:28:05.0492257Z npm ERR! 
2020-03-26T12:28:05.0492839Z npm ERR! Failed at the office-ui-fabric-react-repo@7.0.0 publish:beachball script.
```

--- a/change/beachball-2020-03-26-17-18-49-manual-recovery.json
+++ b/change/beachball-2020-03-26-17-18-49-manual-recovery.json
@@ -1,6 +1,6 @@
 {
   "type": "patch",
-  "comment": "Improve manual recovery message if some packages succeeded",
+  "comment": "Improve manual recovery message if some packages succeeded; increase maxBuffer for publish",
   "packageName": "beachball",
   "email": "elcraig@microsoft.com",
   "dependentChangeType": "patch",

--- a/change/beachball-2020-03-26-17-18-49-manual-recovery.json
+++ b/change/beachball-2020-03-26-17-18-49-manual-recovery.json
@@ -1,0 +1,8 @@
+{
+  "type": "patch",
+  "comment": "Improve manual recovery message if some packages succeeded",
+  "packageName": "beachball",
+  "email": "elcraig@microsoft.com",
+  "dependentChangeType": "patch",
+  "date": "2020-03-27T00:18:49.686Z"
+}

--- a/packages/beachball/src/packageManager/npm.ts
+++ b/packages/beachball/src/packageManager/npm.ts
@@ -1,13 +1,10 @@
-import { spawnSync } from 'child_process';
+import { spawnSync, SpawnSyncOptions } from 'child_process';
 import os from 'os';
-export function npm(
-  args: string[],
-  options?: {
-    cwd: string;
-  }
-) {
+
+export function npm(args: string[], options: SpawnSyncOptions = {}) {
   const npmCmd = os.platform() === 'win32' ? 'npm.cmd' : 'npm';
-  const results = spawnSync(npmCmd, args, options);
+  const results = spawnSync(npmCmd, args, { maxBuffer: 1024 * 1024, ...options });
+
   if (results.status === 0) {
     return {
       stderr: results.stderr.toString().trim(),

--- a/packages/beachball/src/publish/displayManualRecovery.ts
+++ b/packages/beachball/src/publish/displayManualRecovery.ts
@@ -1,8 +1,24 @@
 import { BumpInfo } from '../types/BumpInfo';
-export function displayManualRecovery(bumpInfo: BumpInfo) {
+
+export function displayManualRecovery(bumpInfo: BumpInfo, succeededPackages: Set<string> = new Set<string>()) {
   console.error('Something went wrong with the publish! Manually update these package and versions:');
+  const succeededLines: string[] = [];
+
   bumpInfo.modifiedPackages.forEach(pkg => {
     const packageInfo = bumpInfo.packageInfos[pkg];
-    console.error(`- ${packageInfo.name}@${packageInfo.version}`);
+    const entry = `- ${packageInfo.name}@${packageInfo.version}`;
+    if (succeededPackages.has(packageInfo.name)) {
+      succeededLines.push(entry);
+    } else {
+      console.error(entry);
+    }
   });
+
+  if (succeededLines.length) {
+    console.warn(
+      'These packages and versions were successfully published, but may be invalid due to depending on ' +
+        'package versions for which publishing failed:'
+    );
+    succeededLines.forEach(console.warn);
+  }
 }

--- a/packages/beachball/src/publish/publishToRegistry.ts
+++ b/packages/beachball/src/publish/publishToRegistry.ts
@@ -4,15 +4,17 @@ import { BeachballOptions } from '../types/BeachballOptions';
 import { packagePublish } from '../packageManager/packagePublish';
 import { validatePackageVersions } from './validatePackageVersions';
 import { displayManualRecovery } from './displayManualRecovery';
-import { getNewPackages } from './getNewPackages';
+
 export function publishToRegistry(bumpInfo: BumpInfo, options: BeachballOptions) {
   const { registry, tag, token, access } = options;
   const { modifiedPackages, newPackages } = bumpInfo;
 
   performBump(bumpInfo, options);
 
+  const succeededPackages = new Set<string>();
+
   if (!validatePackageVersions(bumpInfo, registry)) {
-    displayManualRecovery(bumpInfo);
+    displayManualRecovery(bumpInfo, succeededPackages);
     console.error('No packages have been published');
     process.exit(1);
   }
@@ -33,11 +35,11 @@ export function publishToRegistry(bumpInfo: BumpInfo, options: BeachballOptions)
       const result = packagePublish(packageInfo, registry, token, tag, access);
       if (result.success) {
         console.log('Published!');
+        succeededPackages.add(pkg);
       } else {
-        displayManualRecovery(bumpInfo);
+        displayManualRecovery(bumpInfo, succeededPackages);
         console.error(result.stderr);
         process.exit(1);
-        return;
       }
     } else {
       console.warn(

